src/pages/scientificServices/imputation/views/JobHistory.tsx

```diff
@@ -1,20 +1,265 @@
+      cellRenderer: ({ rowIndex }) => {
+        return <ActionCell pipelineRun={paginatedRuns[rowIndex]} />;
+      },
+      size: { basis: 100 },
+    },
+  ];
+};
+
+interface CellProps {
+  // I have absolutely no clue why TS thinks this is unused
+  // eslint-disable-next-line react/no-unused-prop-types
+  pipelineRun: PipelineRun;
+}
+
+const JobIdCell = ({ pipelineRun }: CellProps): ReactNode => {
+  return (
+    <div style={{ display: 'flex', flexDirection: 'column', gap: '0.25rem', width: '100%' }}>
+      <div style={{ width: '100%', overflow: 'hidden' }}>
+        <TooltipCell
+          tooltip={pipelineRun.jobId}
+          style={{
+            whiteSpace: 'nowrap',
+            overflow: 'hidden',
+            textOverflow: 'ellipsis',
+            width: '100%',
+          }}
+        >
+          {pipelineRun.jobId}
+        </TooltipCell>
+      </div>
+      <div
+        style={{
+          width: 'fit-content',
+          fontWeight: 600,
+          backgroundColor: '#4D72AA4D',
+          padding: '0.33rem',
+          borderRadius: '4px',
+          fontSize: '10px',
+        }}
+      >
+        {pipelineRun.pipelineName}
+      </div>
+    </div>
+  );
+};
+
+const DescriptionCell = ({ pipelineRun }: CellProps): ReactNode => {
+  // descriptions can be long, so truncate them and allow the user to hover over them to see the full description
+  return <TooltipCell tooltip={null}>{pipelineRun.description}</TooltipCell>;
+};
+
+const StatusCell = ({ pipelineRun }: CellProps): ReactNode => {
+  return <div style={{ display: 'flex', alignItems: 'center' }}>{getRunStatusIcon(pipelineRun.status)}</div>;
+};
+
+const SubmittedCell = ({ pipelineRun }: CellProps): ReactNode => {
+  return <div>{new Date(pipelineRun.timeSubmitted).toLocaleDateString()}</div>;
+};
+
+const CompletedCell = ({ pipelineRun }: CellProps): ReactNode => {
+  return <div>{pipelineRun.timeCompleted ? new Date(pipelineRun.timeCompleted).toLocaleDateString() : ''}</div>;
+};
+
+const QuotaUsedCell = (props: CellProps): ReactNode => {
+  return <div>{props.pipelineRun.quotaConsumed ? props.pipelineRun.quotaConsumed : 'N/A'}</div>;
+};
+
+const ActionCell = ({ pipelineRun }: CellProps): ReactNode => {
+  // TODO these actions will be implemented in a later ticket
+  return (
+    <div>
+      {pipelineRun.status === 'SUCCEEDED' && (
+        // eslint-disable-next-line jsx-a11y/anchor-is-valid
+        <a href='#' style={{ color: '#46A3E9', fontWeight: 700, textDecoration: 'underline' }}>
+          Download Output
+        </a>
+      )}
+      {pipelineRun.status === 'FAILED' && (
+        // eslint-disable-next-line jsx-a11y/anchor-is-valid
+        <a href='#' style={{ color: '#46A3E9', fontWeight: 700, textDecoration: 'underline' }}>
+          See Details
+        </a>
+      )}
+    </div>
+  );
+};
+
+const getRunStatusIcon = (status: PipelineRunStatus): ReactNode => {
+  switch (status) {
+    case 'SUCCEEDED':
+      return (
```

Comment:
Using text from [mockup](https://www.figma.com/design/xMS9OAHL39AlqDjCI7O4rQ/Imputation-UX-mocks?node-id=597-4411&m=dev):

```suggestion
    case 'RUNNING': 
      return (
        <div style={{ display: 'flex', alignItems: 'center' }}>
          In Progress
        </div>
      );
    case 'SUCCEEDED':
      return (
```

(Or perhaps encode the text in an object constant, and simply reference that in the `default` block.)